packages/flutter/lib/src/material/progress_indicator.dart

```diff
@@ -580,29 +585,44 @@ class _CircularProgressIndicatorState extends State<CircularProgressIndicator> w
   }
 
   Widget _buildMaterialIndicator(BuildContext context, double headValue, double tailValue, double offsetValue, double rotationValue) {
+    final ProgressIndicatorThemeData defaults = Theme.of(context).useMaterial3
+      ? _CircularProgressIndicatorDefaultsM3(context)
+      : _CircularProgressIndicatorDefaultsM2(context);
     final Color? trackColor = widget.backgroundColor ?? ProgressIndicatorTheme.of(context).circularTrackColor;
 
-    return widget._buildSemanticsWrapper(
-      context: context,
-      child: Container(
-        constraints: const BoxConstraints(
-          minWidth: _kMinCircularProgressIndicatorSize,
-          minHeight: _kMinCircularProgressIndicatorSize,
-        ),
-        child: CustomPaint(
-          painter: _CircularProgressIndicatorPainter(
-            backgroundColor: trackColor,
-            valueColor: widget._getValueColor(context),
-            value: widget.value, // may be null
-            headValue: headValue, // remaining arguments are ignored if widget.value is not null
-            tailValue: tailValue,
-            offsetValue: offsetValue,
-            rotationValue: rotationValue,
-            strokeWidth: widget.strokeWidth,
-          ),
+    Widget progressIndicator = Container(
+      constraints: const BoxConstraints(
```

Comment:
Perhaps you could specify `alignment: Alignment.center` here and make constraints a property of the theme. Then the M3 value for the constraints would be tight and the M2 constraints would be as specified here.

Reply:
Thanks for the suggestion! The M3 size is used to make sure that the ProgressIndicator is 48x48, the circle itself should still be 36x36:)